api/src/auth/api_gateway_key_auth.py

```diff
@@ -0,0 +1,78 @@
+import logging
+
+from sqlalchemy import select
+from sqlalchemy.orm import selectinload
+
+import src.util.datetime_util as datetime_util
+from src.adapters import db
+from src.adapters.db import flask_db
+from src.api.route_utils import raise_flask_error
+from src.auth.jwt_user_http_token_auth import JwtUserHttpTokenAuth
+from src.db.models.user_models import UserApiKey
+from src.logging.flask_logger import add_extra_data_to_current_request_logs
+
+logger = logging.getLogger(__name__)
+
+# Initialize the authorization context for API Gateway key authentication
+# This uses the X-API-Key header which is the standard header that AWS API Gateway
+# forwards when api_key_required is set to true
+api_gateway_key_auth = JwtUserHttpTokenAuth(
+    "ApiKey", header="X-API-Key", security_scheme_name="ApiGatewayKeyAuth"
```

Comment:
This shouldn't be a `JwtUserHttpTokenAuth` as it's not using a JWT.


```suggestion
api_gateway_key_auth = HttpTokenAuth(
    "ApiKey", header="X-API-Key", security_scheme_name="ApiGatewayKeyAuth"
```